### PR TITLE
Added warning when attempting to add same mobject as child twice

### LINKED ISSUE #1027: Repeating Matrix objects only show once, when using VGroup
## Description of bug / unexpected behavior

I found this weird, but I am not sure whether this is correct or wrong approach
when I use `VGroup` to group the same `Matrix` objects more than one, it looks that only the last one is appearing, and the other one not, its place is empty

## Expected behavior

it works for me when I add `.cop()` method to  like so, 

## How to reproduce the issue

<details><summary>Code for reproducing the problem</summary>

```py
matrix_r = Matrix([["x"], ["y"]])

lambda_condition_eq = VGroup(
            MathTex("Mr = \lambda r"),
            right_arrow,
            matrix_M,
            matrix_r,
            equal_sign,
            MathTex(r"\lambda"),
            matrix_r,
        ).arrange(RIGHT)
```
here the last `matrix_r` is only appeared 

```py
matrix_r = Matrix([["x"], ["y"]])

adding `.copy()`
lambda_condition_eq = VGroup(
            MathTex("Mr = \lambda r"),
            right_arrow,
            matrix_M,
            matrix_r.copy(),
            equal_sign,
            MathTex(r"\lambda"),
            matrix_r,
        ).arrange(RIGHT)
```
</details>

## Additional media files

<details><summary>Images/GIFs</summary>

originally 
Screen Shot 2021-02-13 at 17 22 17

with `.copy()`
Screen Shot 2021-02-13 at 17 22 54

</details>

## Logs
<details><summary>Terminal output</summary>

```
PASTE HERE OR PROVIDE LINK TO https://pastebin.com/ OR SIMILAR
```

</details>

## System specifications

<details><summary>System Details</summary>

- OS (macOS Big Sur 11.1):
- RAM: 8GB
- Python version (`python/py/python3 --version`): 3.9
- Installed modules (provide output from `pip list`):
```
PASTE HERE
```
</details>

<details><summary>LaTeX details</summary>

+ LaTeX distribution (e.g. TeX Live 2020):
+ Installed LaTeX packages:

</details>

<details><summary>FFMPEG</summary>

Output of `ffmpeg -version`:

```
PASTE HERE
```
</details>

### PR DESCRIPTION
**Pull request by cpark50**
## Overview: What does this pull request change?

Closes #1027

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/mobject/types/vectorized_mobject.py [hunk 1]
```diff
@@ -1755,9 +1755,18 @@ def construct(self):
                     self.play( # Animate group without component
                         (gr-circle_red).animate.shift(RIGHT)
                     )
+
+        Notes
+        -----
+        When adding same mobject more than once, the repeated mobject is ignored.
+        Use .copy() to avoid omission of the repeated mobject.
```
Comment (behackl) [inline manim/mobject/types/vectorized_mobject.py:1762]:
```suggestion
        Use :meth:`.Mobject.copy` to create a separate copy which can then
        be added to the group.
```

File: manim/mobject/types/vectorized_mobject.py [hunk 2]
```diff
@@ -1755,9 +1755,18 @@ def construct(self):
...
         """
         if not all(isinstance(m, (VMobject, OpenGLVMobject)) for m in vmobjects):
             raise TypeError("All submobjects must be of type VMobject")
+        if any(vmobjects.count(elem) > 1 for elem in list(vmobjects)):
+            logger.warning(
+                f"Warning: The same Mobject was added to a Group more than once. Repeated adds are ignored.",
```
Comment (behackl) [inline manim/mobject/types/vectorized_mobject.py:1768]:
```suggestion
                f"Attempted adding some Mobject to a group more than once, this is not possible. Repetitions are ignored.",
```

**Review by behackl — COMMENTED** [review]
Hey! Thank you for your contribution, and sorry it took a while until the review. Your changes look good to me, I do have two suggestions for language improvements; please have a look at them!

File: manim/mobject/types/vectorized_mobject.py [hunk 3]
```diff
@@ -1755,9 +1755,19 @@ def construct(self):
                     self.play( # Animate group without component
                         (gr-circle_red).animate.shift(RIGHT)
                     )
+
+        Notes
+        -----
+        When adding same mobject more than once, the repeated mobject is ignored.
```
Comment (behackl) [inline manim/mobject/types/vectorized_mobject.py:1761]:
```suggestion
        When adding the same mobject more than once, repetitions are ignored.
```

**Review by behackl — APPROVED** [review]
I think the current state of this is fine, although not exactly the approach discussed in #1027. It would be good to hear someone else's opinion about this.

**Review by Darylgolden — APPROVED** [review]
I think I also like this approach.

**Review by Darylgolden — COMMENTED** [review]
Is this problem limited to `VMobject`s, or does it also apply to `Mobject`s more generally?

**Comment by behackl** [comment]
> Is this problem limited to `VMobject`s, or does it also apply to `Mobject`s more generally?

The check should be moved to `Mobject.add`, indeed.

**Comment by Darylgolden** [comment]
> The check should be moved to `Mobject.add`, indeed.

And presumably, `OpenGLMobject.add` too?

**Comment by MrDiver** [comment]
Can we resolve conflicts on this ?

**Comment by behackl** [comment]
> Can we resolve conflicts on this ?

Done, also added a doctest and moved the check from `VGroup` to both `Mobejct` and `OpenGLMobject`. Ready for re-review.

**Review by MrDiver — APPROVED** [review]
LGTM!

**Pull request merged**